src/frontend/src/instance.rs

```diff
@@ -592,7 +624,12 @@ impl SqlQueryHandler for Instance {
         stmt: Statement,
         query_ctx: QueryContextRef,
     ) -> server_error::Result<Output> {
-        self.query_statement(stmt, query_ctx).await
+        let query_interceptor = self.plugins.get::<SqlQueryInterceptorRef>();
+        let stmt = query_interceptor.post_parsing(stmt, query_ctx.clone())?;
```

Comment:
I think `post_parsing` shouldn't be called here, there's no "parsing". If to be paired with `post_execute`, it should be `pre_execute`.